hu.elte.refjava.lang/src/hu/elte/refjava/api/ClassRefactoring.xtend

```diff
@@ -0,0 +1,255 @@
+			}
+			target = matcher.modifiedTarget.toList
+			bindings.putAll(matcher.bindings)
+			
+			return true
+		} catch (Exception e) {
+			println(e)
+			return false
+		}
+	}
+	
+	def protected void setMetaVariables() {
+		//empty
+	}
+	
+	def protected safeTargetCheck() {
+		return true
+	}
+	
+	def protected targetCheck(String targetPatternString) {
+		try {
+			this.targetString = targetPatternString
+			if(!matcher.match(PatternParser.parse(targetPatternString), target, targetTypeReferenceString)) {
+				return false
+			}
+			bindings.putAll(matcher.bindings)
+			return true
+		} catch (Exception e) {
+			println(e)
+			return false
+		}	
+	}
+
+	def private safeCheck() {
+		try {
+			return check()
+		} catch (Exception e) {
+			println(e)
+			return false
+		}
+	}
+
+	def protected check() {
+		//TODO
+		if(replacementString != "-") {
+			//return Check.isUniqueMethodIn(target, Check.enclosingClass(target))
+			
+			return true
+			
+			
+		} else {
+			if(target.head instanceof MethodDeclaration) {
+				return Check.hasSuperClass(Check.enclosingClass(target))
+					&& if (Check.isPrivate(target)) Check.references(target, Check.enclosingClass(target)).empty else true
+					&& Check.accessedFieldsOfEnclosingClass(target, Check.enclosingClass(target)).empty
+					&& Check.accessedMethodsOfEnclosingClass(target, Check.enclosingClass(target)).empty
+				//	&& Check.isUniqueMethodIn(target, Check.superClass(Check.enclosingClass(target)))
+					
+				
+			} else if (target.head instanceof FieldDeclaration) {
+				return Check.hasSuperClass(Check.enclosingClass(target))
+					&& if (Check.isPrivate(target)) Check.references(target, Check.enclosingClass(target)).empty else true
+				//	&& Check.isUniqueFieldIn(target, Check.superClass(Check.enclosingClass(target)))
+			}
+			return false
+		}
+	}
+
+	def private safeBuild() {
+		try {
+			if(replacementString != "-") {
+				replacement = builder.build(target.head.AST, bindings, nameBindings, typeBindings, parameterBindings, replacementTypeReferenceString)
+			}
+		} catch (Exception e) {
+			println(e)
+			return false
+		}
+		return true
+	}
+
+	def private safeReplace() {
+		try {
+			
+			//we only need to replace if we creating a new method
+			if (replacement !== null) {
+				val rewrite = builder.rewrite
+				target.tail.forEach[rewrite.remove(it, null)]
+			
+				val group = rewrite.createGroupNode( replacement )
+				rewrite.replace( target.head, group, null)
+				var edits = rewrite.rewriteAST(document, null)
+				edits.apply(document)
+			}
+			
+			val compUnit = Utils.getCompilationUnit(target.head)
+			val targetICompUnit = compUnit.getJavaElement() as ICompilationUnit
+			
+			val parser = ASTParser.newParser(AST.JLS12);
+			parser.source = targetICompUnit;
+			val targetCompUnit = parser.createAST(null) as CompilationUnit;
+			
+			targetCompUnit.recordModifications
+			
+			if (replacementString == "-" && definitionString == "target" && matchingString == "target") {
```

Comment:
I would strongly suggest to parse these strings as soon as possible and use the resulting ASTs instead later on. This abstraction gap does no good for maintainability.